src/Engine/Events/EventIR.cpp

```diff
@@ -838,362 +838,365 @@ std::string EventIR::toString() const {
             break;
-        case EVENT_ToggleActorFlag:
+        case EVENT_ToggleActorFlag:  // TODO(yoctozepto): not present in used MM7 data
             requireSize(14);
-            ir.data.actor_flag_descr.id = EVT_DWORD(&evt->v5);
-            ir.data.actor_flag_descr.attr = static_cast<ActorAttribute>(EVT_DWORD(&evt->v9));
-            ir.data.actor_flag_descr.is_set = evt->v13;
+            ir.data.actor_flag_descr.id = sbr.read<uint32_t>();
+            ir.data.actor_flag_descr.attr = static_cast<ActorAttribute>(sbr.read<uint32_t>());
+            ir.data.actor_flag_descr.is_set = sbr.read<uint8_t>();
             break;
         case EVENT_RandomGoTo:
             requireSize(11);
-            ir.data.random_goto_descr.random_goto[0] = evt->v5;
-            ir.data.random_goto_descr.random_goto[1] = evt->v6;
-            ir.data.random_goto_descr.random_goto[2] = evt->v7;
-            ir.data.random_goto_descr.random_goto[3] = evt->v8;
-            ir.data.random_goto_descr.random_goto[4] = evt->v9;
-            ir.data.random_goto_descr.random_goto[5] = evt->v10;
-            ir.data.random_goto_descr.random_goto_len = 1 + !!evt->v6 + !!evt->v7 + !!evt->v8 + !!evt->v9 + !!evt->v10;
+            {
+                auto &rgt = ir.data.random_goto_descr;
+                rgt.random_goto_len = 0;
+                for (int i = 0; i < rgt.random_goto.size(); i++) {
+                    rgt.random_goto[i] = sbr.read<uint8_t>();
+                    if (rgt.random_goto[i] > 0) {
+                        rgt.random_goto_len++;
+                    }
+                }
+                if (rgt.random_goto_len == 0) {
+                    throw Exception("RandomGoTo event has 0 targets");
+                }
+            }
             break;
-        case EVENT_InputString:
+        case EVENT_InputString:  // TODO(yoctozepto): not present in used MM7 data
             requireSize(9);
-            ir.data.text_id = EVT_DWORD(&evt->v5);
+            ir.data.text_id = sbr.read<uint32_t>();
             break;
         case EVENT_StatusText:
             requireSize(9);
-            ir.data.text_id = EVT_DWORD(&evt->v5);
+            ir.data.text_id = sbr.read<uint32_t>();
             break;
         case EVENT_ShowMessage:
             requireSize(9);
-            ir.data.text_id = EVT_DWORD(&evt->v5);
+            ir.data.text_id = sbr.read<uint32_t>();
             break;
         case EVENT_OnTimer:
-            requireSize(13);
-            ir.data.timer_descr.is_yearly = evt->v5;
-            ir.data.timer_descr.is_monthly = evt->v6;
-            ir.data.timer_descr.is_weekly = evt->v7;
-            ir.data.timer_descr.daily_start_hour = evt->v8;
-            ir.data.timer_descr.daily_start_minute = evt->v9;
-            ir.data.timer_descr.daily_start_second = evt->v10;
-            ir.data.timer_descr.alt_halfmin_interval = evt->v11 + (evt->v12 << 8);
+        case EVENT_OnLongTimer:
+            requireSize(15);
```

Comment:
Was 13 originally. Is this intentional?

Reply:
Yup. It seems there is always some extra data word. I am planning to check if it bears any meaning but it's there and we now read it and discard so the "required" size for this type grows by 2. :-)